tests/cross/src/test/scala/tests/pc/CompletionArgSuite.scala

```diff
@@ -163,16 +163,10 @@ class CompletionArgSuite extends BaseCompletionSuite {
         |  Option[Int](@@)
         |}
         |""".stripMargin,
-    """|x = : Int
+    """|x = : A
```

Comment:
It is. In this test it falls to `methodSym.decodedName == "apply"` case (`ArgCompletions` line `27`). If I delete that case in `case-class-apply1` I will only get `foo = : Int` sice it will find the `apply` method in the object `A` and won't find constructor of the `A` class. So I decided it's better to have this type less precise.

Reply:
It doesn't fail `methodSym.decodedName == "apply"`, on the contrary this condition is true in this case.

In that test before changes in this PR `method.tpe.paramss.headOption` was used to collect arg completions.
However, if using `method.tpe.paramss.headOption` doesn't work for constructors (synthetic applies) but it will work for `apply` method defined in a companion object (non-synthetic).  So in case of `case-class-apply1` it only found the `apply` explicitly defined in the object and did not find the synthetic one. For this reason I handle `apply` also using `getParamsFromObject`. In `getParamsFromObject` we collect matching methods manually so we don't get the same narrowing of the method type we have for `method.tpe.paramss.headOption`. That's the source of regression.

Reply:
We should be able to improve that later on with something like `metalsSeenFromType` method, but it's fine to leave that as an issue also. Thanks!